src/main/java/soconnect/logic/commands/tagcommands/TagRemoveCommand.java

```diff
@@ -0,0 +1,102 @@
+package soconnect.logic.commands.tagcommands;
+
+import static java.util.Objects.requireNonNull;
+import static soconnect.commons.util.CollectionUtil.requireAllNonNull;
+import static soconnect.logic.parser.CliSyntax.PREFIX_TAG;
+import static soconnect.model.Model.PREDICATE_SHOW_ALL_PERSONS;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import soconnect.commons.core.Messages;
+import soconnect.commons.core.index.Index;
+import soconnect.logic.commands.CommandResult;
+import soconnect.logic.commands.exceptions.CommandException;
+import soconnect.model.Model;
+import soconnect.model.person.Address;
+import soconnect.model.person.Email;
+import soconnect.model.person.Name;
+import soconnect.model.person.Person;
+import soconnect.model.person.Phone;
+import soconnect.model.tag.Tag;
+
+/**
+ * Removes a tag from a contact.
+ */
+public class TagRemoveCommand extends TagCommand {
+
+    public static final String COMMAND_WORD = "remove";
+
+    public static final String MESSAGE_USAGE = TagCommand.COMMAND_WORD + " "
+            + COMMAND_WORD + ": Removes a tag from the contact "
+            + "by the index number used in the displayed person list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + PREFIX_TAG + "TAG...\n"
+            + "Example: " + TagCommand.COMMAND_WORD + " "
+            + COMMAND_WORD + " 1 "
+            + PREFIX_TAG + "owesMoney";
+
+    public static final String MESSAGE_REMOVE_TAG_SUCCESS = "Tag removed: %1$s";
+    public static final String MESSAGE_NO_SUCH_TAG = "This tag does not exist";
+    public static final String MESSAGE_NO_TAG = "Please specify a tag";
```

Comment:
I notice a lot of repeated `MESSAGE` here, optimization I can think of is to compile these messages in one file and import from it. That way if we need to make changes to the messages we only need to change in one place :D

Reply:
I see that the original ab3 also kept their `MESSAGE` in their respective classes. I feel that it is easier to track since I do use the same names but with different strings printed differently from each other.